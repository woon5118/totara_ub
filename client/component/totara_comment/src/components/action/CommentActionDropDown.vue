<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Kian Nguyen <kian.nguyen@totaralearning.com>
  @module totara_comment
-->
<template>
  <Dropdown position="bottom-right" class="tui-commentActionDropDown">
    <template v-slot:trigger="{ toggle, isOpen }">
      <ButtonIcon
        v-if="showActions"
        :aria-expanded="isOpen ? 'true' : 'false'"
        :aria-label="$str('triggermenu', 'totara_comment')"
        :styleclass="{ transparentNoPadding: true }"
        @click.prevent="toggle"
      >
        <More size="300" />
      </ButtonIcon>
    </template>

    <DropdownItem v-if="showUpdateAction" @click="$emit('click-edit')">
      {{ $str('edit', 'core') }}
    </DropdownItem>

    <DropdownItem v-if="showDeleteAction" @click="$emit('click-delete')">
      {{ $str('delete', 'core') }}
    </DropdownItem>

    <DropdownItem
      v-if="showReportAction"
      @click="$emit('click-report-content')"
    >
      {{ $str('reportcontent', 'totara_comment') }}
    </DropdownItem>
  </Dropdown>
</template>

<script>
import Dropdown from 'tui/components/dropdown/Dropdown';
import More from 'tui/components/icons/More';
import DropdownItem from 'tui/components/dropdown/DropdownItem';
import ButtonIcon from 'tui/components/buttons/ButtonIcon';

export default {
  components: {
    ButtonIcon,
    Dropdown,
    More,
    DropdownItem,
  },

  props: {
    showDeleteAction: {
      type: Boolean,
      default: true,
    },

    showUpdateAction: {
      type: Boolean,
      default: true,
    },

    showReportAction: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    showActions() {
      return (
        this.showDeleteAction || this.showUpdateAction || this.showReportAction
      );
    },
  },
};
</script>

<lang-strings>
  {
    "core": [
      "delete",
      "edit"
    ],

    "totara_comment": [
      "triggermenu",
      "reportcontent"
    ]
  }
</lang-strings>
