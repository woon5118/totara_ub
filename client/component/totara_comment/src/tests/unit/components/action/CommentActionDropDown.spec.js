/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Kian Nguyen <kian.nguyen@totaralearning.com>
 * @module totara_comment
 */

import CommentActionDropDown from 'totara_comment/components/action/CommentActionDropDown';
import { shallowMount } from '@vue/test-utils';

describe('totara_comment/components/action/CommentActionDropDown.vue', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(CommentActionDropDown, {
      propsData: {
        deleteAble: true,
        editAble: true,
        reportAble: false,
      },
      mocks: {
        $str(x, y) {
          return `${x}-${y}`;
        },
      },
    });
  });

  it('Checks snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
