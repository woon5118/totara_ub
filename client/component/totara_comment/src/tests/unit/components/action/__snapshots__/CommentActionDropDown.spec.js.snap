// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`totara_comment/components/action/CommentActionDropDown.vue Checks snapshot 1`] = `
<dropdown-stub
  animation="default"
  canclose="true"
  class="tui-commentActionDropDown"
  closeonclick="true"
  contextmode="contained"
  deleteable="true"
  editable="true"
  position="bottom-right"
  role="menu"
  separator="true"
>
   
  <dropdownitem-stub
    role="menuitem"
  >
    
    edit-core
  
  </dropdownitem-stub>
   
  <dropdownitem-stub
    role="menuitem"
  >
    
    delete-core
  
  </dropdownitem-stub>
   
  <dropdownitem-stub
    role="menuitem"
  >
    
    reportcontent-totara_comment
  
  </dropdownitem-stub>
</dropdown-stub>
`;
